safe_core/src/client/mock/tests.rs

```diff
@@ -1257,6 +1362,7 @@ fn auth_actions_from_app() {
     send_req_expect_failure!(
         &mut app_conn_manager,
         &app_key,
+        None,
```

Comment:
Requests by app would need tokens here. The `AccessDenied` returned here is meant to be checked for a different scenario(Access being denied due to reading data, deleting data etc). Failure due to tokens must be differentiated from this.

Reply:
how should we differentiaite? 

From your explanation reads like access isnt denied, just that data may not exist?

Reply:
So for example, `DeleteMdata` is a user/client only operation. `AccessDenied` is responded when Apps try the same. If we do not pass Tokens here, we'd be thrown a `AccessDenied` as well. We cannot verify from which operation we have the error. So, we should be testing token errors separately and, we should assert them to have `InvalidToken` error or something similar.

Reply:
`AccessDenied` is vague. Why is it being denied here? Because it's an app? Or because it doesn't have permission for that piece of data?

Hmm, `AccessDenied` is the _same_ as a token not being signed (in terms of the result). Or not being current...
I think we need a new error for this app case vs auth as well.

`TokenMissing` makes sense.
`TokenSignatureInvalid` when not signed.
`TokenCaveatInvalid` when a caveat fails (perhaps with reject reason)? 

more?

Reply:
>Why is it being denied here? Because it's an app?

Yup, it is being denied _here_ as Apps cannot delete data. 

> Or because it doesn't have permission for that piece of data?

It could for any operation that they do not for permissions to.(insert/ update/ listauthkeys etc.)
So having new errors would definitely help here.

>TokenMissing makes sense.
TokenSignatureInvalid when not signed.
TokenCaveatInvalid when a caveat fails

All the three sound good to me :)
`TokenNotCurrent` will be one, if the app provides older tokens.